autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/pixel/AutofillPixelInterceptor.kt

```diff
@@ -0,0 +1,123 @@
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SAVE_LOGIN_PROMPT_SHOWN
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SAVE_PASSWORD_PROMPT_DISMISSED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SAVE_PASSWORD_PROMPT_SAVED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SAVE_PASSWORD_PROMPT_SHOWN
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_AUTOPROMPT_DISMISSED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_AUTOPROMPT_SELECTED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_AUTOPROMPT_SHOWN
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_PROMPT_DISMISSED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_PROMPT_SELECTED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.AUTOFILL_SELECT_LOGIN_PROMPT_SHOWN
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.MENU_ACTION_AUTOFILL_PRESSED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelNames.SETTINGS_AUTOFILL_MANAGEMENT_OPENED
+import com.duckduckgo.autofill.impl.pixel.AutofillPixelParameters.AUTOFILL_DEFAULT_STATE
+import com.duckduckgo.autofill.store.AutofillPrefsStore
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import okhttp3.Interceptor
+import okhttp3.Interceptor.Chain
+import okhttp3.Response
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = PixelInterceptorPlugin::class,
+)
+class AutofillPixelInterceptor @Inject constructor(
+    private val autofillStore: AutofillPrefsStore,
+) : PixelInterceptorPlugin, Interceptor {
+
+    private fun isInPixelsList(pixel: String): Boolean {
+        return pixels.firstOrNull { pixel.startsWith(it.pixelName) } != null
+    }
+
+    override fun intercept(chain: Chain): Response {
+        val request = chain.request().newBuilder()
+        val pixel = chain.request().url.pathSegments.last().removeSuffixes()
+
+        val url = if (isInPixelsList(pixel)) {
+            val defaultState = autofillStore.wasDefaultStateEnabled()
+            chain.request().url.newBuilder().addQueryParameter(AUTOFILL_DEFAULT_STATE, defaultState.asDefaultStateParam()).build()
+        } else {
+            chain.request().url
+        }
+
+        return chain.proceed(request.url(url).build())
+    }
+
+    override fun getInterceptor(): Interceptor = this
+
+    private fun Boolean.asDefaultStateParam(): String {
+        return if (this) "on" else "off"
+    }
+
+    private fun String.removeSuffixes(): String {
+        return this
+            .removeSuffix("_android_phone")
+            .removeSuffix("_android_tablet")
+    }
+
+    companion object {
+
+        val pixels = listOf(
```

Comment:
Nit: Could also just be a check for starts with `m_autofill_` or `m_nav_autofill` + exceptions. But this approach you have also works.